src/clusterfuzz/_internal/build_management/build_manager.py

```diff
@@ -1671,3 +1673,11 @@ def check_app_path(app_path='APP_PATH'):
   # APP_PATH.
   return (not environment.get_value('APP_NAME') or
           environment.get_value(app_path))
+
+
+def get_bucket_path(name):
+  """Return build bucket path, applying any set overrides."""
+  bucket_path = environment.get_value(name)
+  if environment.is_android():
```

Comment:
Let's do this in a more generic way (not specific to android), and do something like (as described in https://github.com/google/clusterfuzz/pull/2410/files#discussion_r690874921):

```python
if os.path.basename(bucket_path) == PLATFORM_ID_URLS_FILENAME:
   _apply_platform_id_overrides(environment.get_platform_id(), bucket_path):

def _apply_platform_id_overrides(...):
  # read the `bucket_path`, parse as JSON, and map based on platform_id. 
```

Reply:
agreed. This seems like a better idea.

Reply:
What do you think of this strategy:
set the bucket path to the config.json file url.
then, based on your suggestion above, at run time, if os.path.basename is the config file, try to get the actual path based on the device platform id.

Reply:
Yes, this is exactly what I'm proposing, thanks!